### PR TITLE
feat: Hub submodule - add option to use existing service account to register clusters.

### PR DESCRIPTION
**Pull request by cloud-pharaoh**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/hub/README.md [hunk 1]
```diff
@@ -41,7 +41,9 @@ To deploy this config:
 | gke\_hub\_sa\_name | Name for the GKE Hub SA stored as a secret `creds-gcp` in the `gke-connect` namespace. | string | `"gke-hub-sa"` | no |
 | location | The location (zone or region) this cluster has been created in. | string | n/a | yes |
 | project\_id | The project in which the resource belongs. | string | n/a | yes |
+| sa\_private\_key | Private key for service account. Required only if `use_existing_sa` is set to `true`. | string | `"null"` | no |
```
Comment (bharathkkb) [inline modules/hub/README.md:44]: Should we call out the fact that this should be base64 encoded?

**Review by bharathkkb — APPROVED** [review]
LGTM
Thanks for the PR!

File: modules/hub/README.md (hunk 1 above)
Comment (cloud-pharaoh) [reply modules/hub/README.md:44]: Yes. That is a good idea.

**Pull request merged**